### ISSUE TITLE
Fish Finder

### ISSUE THREAD

**Issue by Gadgeteering** [merbanan/rtl_433#2126]
Anybody decoded a fish finder yet ?

[![](https://m.media-amazon.com/images/I/61woH6hwS4L._AC_AA160_.jpg)](https://www.amazon.co.uk/Lucky-Wireless-Transducer-Compatible-Display/dp/B07GP2XBC8/?_encoding=UTF8&pd_rd_w=XNezJ&content-id=amzn1.sym.822afc68-e2bb-4420-ac6e-1044cd2e7c1d&pf_rd_p=822afc68-e2bb-4420-ac6e-1044cd2e7c1d&pf_rd_r=2DPK2DACNEYA31DPHA3H&pd_rd_wg=JagPw&pd_rd_r=3c254f89-5e3c-4fdc-9971-04a56788674d&ref_=pd_gw_ci_mcx_mi&th=1)

![71ZogLx9cDL _AC_SL1500_](https://user-images.githubusercontent.com/6560476/183294868-cd3be8fa-3bdf-4ca3-88b8-6059039b816a.jpg)

[Internal Photos](https://fccid.io/2AZVB-FF1108-1CWLA)

**Comment by Gadgeteering** [comment]
I have uploaded some samples data
[samples.zip](https://github.com/merbanan/rtl_433/files/9184126/samples.zip)

**Comment by zuckschwerdt** [comment]
`rtl_433 -X 'n=fishfinder,m=OOK_PWM,s=175,l=575,r=700'`

There are three different signals: 20 bits `c0000`, 16 bits `b3cc` or `b3cc` and one long pulse (6.6 ms, shown as `{1}0`).
Data could be in that `b3..` packet, or if those don't vary then the data could be in the timing? Faster signals -> more fish close by? Or what should the signal indicate?

**Comment by Gadgeteering** [comment]
I think the 20bits might be a sync code as this changes when you switch the unit on and off

The 16bits might be the water temperature and the other data packets are the returns

[Sample.zip](https://github.com/merbanan/rtl_433/files/9184387/Sample.zip)

**Comment by Gadgeteering** [comment]
I don't think that it is pure digital signal there maybe an analogue echo signal, the temperature maybe transmitted as digital signal.

[Fish finder sonar: circuits, protocols and transducers](https://www.eevblog.com/forum/projects/fish-finder-sonar-circuits-protocols-and-transducers/)

**Comment by Gadgeteering** [comment]
I have now managed to decode the 20 Bits, the first 16 bits are the temperature, strangely the bits are doubled up, odd and even bits are the same. So I needed to extract every other bit to decode the temperature. I still need to calibrate the algorithm to get the correct temperature.

<img width="1268" alt="Screenshot 2022-08-03 at 19 57 23" src="https://user-images.githubusercontent.com/6560476/182687728-df55b5b4-a0ca-4b05-9562-7dd82136aaed.png">
